### PR TITLE
Kustomize argocd

### PR DESCRIPTION
**Pull request by HumairAK**
## This introduces a breaking change

- [x] Yes
- [ ] No

## This Pull Request implements

- Remove Ansible playbook and jinja templates 
- Convert the argocd deployment to use Kustomize 
- Update Docs to reflect changes 
- Add mkdocs yaml

### CODE REVIEW

File: docs/admin/add_namespace_to_cluster.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Add namespace to cluster
+* sops 3.6+
```
Comment (anishasthana) [inline docs/admin/add_namespace_to_cluster.md:2]: Shouldn't this be under a "prerequisites" header?

Comment (HumairAK) [reply docs/admin/add_namespace_to_cluster.md]: fixed, and amended other instructions in the admin section as well

File: docs/admin/add_namespace_to_cluster.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```yaml
+...
+ 10 stringData:
+ 11     name: apps.ocp.prod.psi.redhat.com
+ 12     config: ...
+ 14     namespaces: namespace-1,namespace-2 # Update this field by appending the new namespace
+ 15     server: ...
+```
+
+Once done, submit a PR with this change.
```
Comment (anishasthana) [inline docs/admin/add_namespace_to_cluster.md:29]: Don't you need to re-encrypt it after you are done editing? (does it automatically save as encrypted?)

File: docs/admin/add_new_cluster_spec.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Adding a new cluster spec
+
+## Pre-requisites
+
+* sops 3.6+
+* imported aicoe-sre gpg key
+
+## Instructions
+
+ArgoCD will need a service account present on the cluster for deployments. Where the SA is located is irrelevant, though it's advised to have it be located in it's own indepenent namespace. For consistency name this service account `argocd-manager`.
```
Comment (anishasthana) [inline docs/admin/add_new_cluster_spec.md:10]: the second `it's` should be `its`

File: docs/admin/update_gpg_key.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+$ gpg --export-secret-keys "${KEY_ID}" | base64 > private.asc
+```
+
+```bash
+# From the repo root
+$ cd manifests/overlays/prod/resources/secrets/gpg
+$ sops secret.enc.yaml
+```
+
+Copy the contents of `private.asc` onto the `private.key` field.
```
Comment (anishasthana) [inline docs/admin/update_gpg_key.md:22]: `into`, not `onto`

File: docs/create_argocd_application_manifest.md [hunk 5]
```diff
@@ -89,13 +92,20 @@ Let's go over what some of the fields in the `example_app.yaml` refer to:
       ...
       ```
 
-NOTE: You may need to disable schema validation if your deployments are failing. This is due to the ArgoCD api validator having a very strict api spec.
+> NOTE: You may need to disable schema validation if your deployments are failing. This is due to the ArgoCD api validator having a very strict api spec.
+
+Once you are done add the application yaml to the Kustomization file by running the follwing _from the repository root directory_.
```
Comment (anishasthana) [inline docs/create_argocd_application_manifest.md:97]: `follwing`

File: docs/admin/add_namespace_to_cluster.md (hunk 2 above)
Comment (HumairAK) [reply docs/admin/add_namespace_to_cluster.md:29]: Yep auto saves as encrypted

**Comment by HumairAK** [comment]
ignore pre-commit failures, the lack of newlines on configs is intentional

**Pull request merged**